src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
```

Comment:
lets explain this in the doc comment.

what is the reason for the order? is this the only way to create a $_POST stream and files is just an additional use of it? what order do implementations usually have?

what would be the semantics of calling this with both arrays empty? does that make any sense or should we throw an exception then?